server/controllers/article/edit-topic.php

```diff
@@ -56,8 +57,11 @@ public function handler() {
         if(Controller::request('icon')) {
             $topic->icon = Controller::request('icon');
         }
+        if(Controller::request('private') || Controller::request('private') == 0) {
```

Comment:
Please check only if it's not null
```suggestion
        if (Controller::request('private') !== null) {
```